tests/integration/src/setup.rs

```diff
@@ -23,17 +23,95 @@ pub use frame_support::{
     traits::{OnFinalize, OnIdle, OnInitialize},
     weights::Weight,
 };
-pub use sp_core::H160;
+pub use pallet_evm::AddressMapping;
+pub use sp_core::{H160, H256, U256};
 pub use sp_runtime::{AccountId32, MultiAddress};
 
+pub use astar_primitives::ethereum_checked::AccountMapping;
+
 #[cfg(feature = "shibuya")]
 pub use shibuya::*;
 #[cfg(feature = "shibuya")]
 mod shibuya {
+    use super::*;
     pub use shibuya_runtime::*;
 
     /// 1 SBY.
     pub const UNIT: Balance = SBY;
+
+    // TODO: once Account Unification is finished, remove `alith` and `alicia`,
+    // which are mocks of two way account/address mapping.
+
+    /// H160 address mapped from `ALICE`.
+    pub fn alith() -> H160 {
+        h160_from(ALICE)
+    }
+
+    /// `AccountId32` mapped from `alith()`.
+    pub fn alicia() -> AccountId32 {
+        account_id_from(alith())
+    }
+
+    /// Convert `H160` to `AccountId32`.
+    pub fn account_id_from(address: H160) -> AccountId32 {
+        <Runtime as pallet_evm::Config>::AddressMapping::into_account_id(address)
+    }
+
+    /// Convert `AccountId32` to `H160`.
+    pub fn h160_from(account_id: AccountId32) -> H160 {
+        <Runtime as pallet_ethereum_checked::Config>::AccountMapping::into_h160(account_id)
+    }
+
+    /// Deploy an EVM contract with code.
+    pub fn deploy_evm_contract(code: &str) -> H160 {
+        assert_ok!(EVM::create2(
+            RuntimeOrigin::root(),
+            alith(),
+            hex::decode(code).expect("invalid code hex"),
+            H256::zero(),
+            U256::zero(),
+            1_000_000,
+            U256::from(DefaultBaseFeePerGas::get()),
+            None,
+            None,
+            vec![],
+        ));
+        match System::events()
+            .iter()
+            .last()
+            .expect("no event found")
+            .event
+        {
+            RuntimeEvent::EVM(pallet_evm::Event::Created { address }) => address,
+            _ => panic!("Deploy failed."),
+        }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `deploy_evm_contract` function is missing error handling for the `assert_ok!` macro. It's recommended to use the `Result` type instead of panicking. Additionally, it would be helpful to add a comment explaining the purpose of this function. Consider the following changes:

```rust
/// Deploy an EVM contract with code.
pub fn deploy_evm_contract(code: &str) -> Result<H160, &'static str> {
    let create_result = EVM::create2(
        RuntimeOrigin::root(),
        alith(),
        hex::decode(code).expect("invalid code hex"),
        H256::zero(),
        U256::zero(),
        1_000_000,
        U256::from(DefaultBaseFeePerGas::get()),
        None,
        None,
        vec![],
    );

    match create_result {
        Ok(()) => {
            let address = match System::events().iter().last().expect("no event found").event {
                RuntimeEvent::EVM(pallet_evm::Event::Created { address }) => address,
                _ => return Err("Deploy failed."),
            };
            Ok(address)
        }
        Err(_) => Err("Create failed."),
    }
}
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->